arrows/vxl/color_commonality_filter.cxx

```diff
@@ -0,0 +1,610 @@
+  }
+
+  if( image_data->depth() != 1 && image_data->depth() != 3 )
+  {
+    LOG_ERROR( logger(), "Invalid number of input planes!" );
+    return image_data;
+  }
+
+  // Get input image
+  vil_image_view_base_sptr view =
+    vxl::image_container::vital_to_vxl( image_data->get_image() );
+
+  // Perform different actions based on input type
+#define HANDLE_CASE(T)                                                 \
+  case T:                                                              \
+    {                                                                  \
+      typedef vil_pixel_format_type_of<T >::component_type pix_t;      \
+      vil_image_view< pix_t > input = view;                            \
+                                                                       \
+      if( d->smooth_image )                                            \
+      {                                                                \
+        vil_image_view< pix_t > smoothed;                              \
+        vil_gauss_filter_2d( input, smoothed, 0.5, 2 );                \
+        input = smoothed;                                              \
+      }                                                                \
+      if( input.nplanes() == 1 )                                       \
+      {                                                                \
+        vil_image_view< pix_t > output;                                \
+        d->settings.resolution_per_channel = d->intensity_resolution;  \
+        d->settings.histogram = &d->intensity_histogram;               \
+        perform_filtering( input, output, d->settings );               \
+        return std::make_shared< vxl::image_container >( output );     \
+      }                                                                \
+      else                                                             \
+      {                                                                \
+        vil_image_view< pix_t > output;                                \
+        d->settings.resolution_per_channel = d->color_resolution_per_chan; \
+        d->settings.histogram = &d->color_histogram;                   \
+        perform_filtering( input, output, d->settings );               \
+        return std::make_shared< vxl::image_container >( output );     \
+      }                                                                \
+    }                                                                  \
+    break;                                                             \
+
+  switch( view->pixel_format() )
+  {
+    HANDLE_CASE(VIL_PIXEL_FORMAT_BOOL);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_BYTE);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_SBYTE);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_UINT_16);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_UINT_32);
+    HANDLE_CASE(VIL_PIXEL_FORMAT_UINT_64);
+#undef HANDLE_CASE
+
+  default:
+    LOG_ERROR( logger(), "Invalid type received" );
+    return image_data;
```

Comment:
This seems like a dangerous return value. At very least, I think this should be consistent with the final unreachable return and the other analogous `filter` functions. And actually, I think if any functions should return the original image, it should be the `convert_image` or `average`, because then there's at lease some chance of it being correct or meaningful. E.g., the current image is closely related to the average of historical frames.